### PR TITLE
CLN: pytables _set_tz/_get_tz

### PR DESCRIPTION
**Pull request by jbrockmendel**
removes preserve_UTC argument, which never did anything, and didnt have a clear purpose

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -4651,37 +4652,39 @@ def _get_info(info, name):
...
-        zone = tz.utcoffset().total_seconds()
     return zone
 
 
-def _set_tz(values, tz, preserve_UTC: bool = False, coerce: bool = False):
+def _set_tz(
+    values: Union[np.ndarray, Index],
+    tz: Optional[Union[str, tzinfo]],
+    coerce: bool = False,
+):
```
Comment (WillAyd) [inline pandas/io/pytables.py]: Is the return easy to annotate here? DTI or Ndarray?

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -4651,37 +4652,39 @@ def _get_info(info, name):
...
-    values : ndarray
-    tz : string/pickled tz object
-    preserve_UTC : bool,
-        preserve the UTC of the result
+    values : ndarray or Index
+    tz : str or tzinfo
     coerce : if we do not have a passed timezone, coerce to M8[ns] ndarray
     """
+    if isinstance(values, DatetimeIndex):
+        # If we values is tzaware, the tz gets dropped in the values.ravel()
```
Comment (WillAyd) [inline pandas/io/pytables.py]: Is `we` a typo?

**Review by WillAyd — COMMENTED** [review]
looks good generally

File: pandas/io/pytables.py [hunk 3]
```diff
@@ -4651,37 +4652,39 @@ def _get_info(info, name):
...
+        #  call below (which returns an ndarray).  So we are only non-lossy
+        #  if `tz` matches `values.tz`.
+        assert values.tz is None or values.tz == tz
+
     if tz is not None:
         name = getattr(values, "name", None)
         values = values.ravel()
```
Comment (jreback) [inline pandas/io/pytables.py]: i wonder if we need the ravel any longer

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/pytables.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: i havent checked if we ever actually pass 2D ndarray, but the ravel is definitely needed in the case where we have a DTI (or we would have to replace it with DTI-specific code) (see the comment a few lines up)

File: pandas/io/pytables.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: yes

File: pandas/io/pytables.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: yes, will do

**Comment by jbrockmendel** [comment]
updated per comments

BTW thanks @jreback and @WillAyd for putting up with the deluge of PRs in this file.  I think once the statefulness issues are addressed it will become much more maintainable long-term

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok sure

**Pull request merged**